src/os/ui/areas.js

```diff
@@ -167,7 +183,106 @@ os.ui.AreasCtrl.prototype.export = function() {
         return null;
       }).filter(os.fn.filterFalsey);
 
-  os.ui.ex.AreaExportCtrl.start(areas);
+  os.ui.ex.AreaExportCtrl.start(formattedAreas);
+};
+
+
+/**
+ * Events fired by the query area import menu.
+ * @enum {string}
+ */
+os.ui.AreasCtrl.EventType = {
+  ALL: 'areaexport:all',
+  SELECTED: 'areaexport:selected',
+  ACTIVE: 'areaexport:active'
+};
+
+
+/**
+ * Opens the area export menu.
+ *
+ * @export
+ */
+os.ui.AreasCtrl.prototype.openExportMenu = function() {
+  var target = this.element.find('.js-export-group');
+  var menu = new os.ui.menu.Menu(new os.ui.menu.MenuItem({
+    type: os.ui.menu.MenuItemType.ROOT,
+    children: [{
+      label: 'Export All',
+      eventType: os.ui.AreasCtrl.EventType.ALL,
+      tooltip: 'Export all areas',
+      handler: os.ui.AreasCtrl.exportEventHandler_,
+      sort: 1
+    }, {
+      label: 'Export Selected',
+      eventType: os.ui.AreasCtrl.EventType.SELECTED,
+      tooltip: 'Export only selected areas',
+      handler: os.ui.AreasCtrl.exportEventHandler_,
+      sort: 2
+    }, {
+      label: 'Export Active',
+      eventType: os.ui.AreasCtrl.EventType.ACTIVE,
+      tooltip: 'Export only active areas',
+      handler: os.ui.AreasCtrl.exportEventHandler_,
+      sort: 3
+    }]
+  }));
+
+  var menuRoot = menu.getRoot();
+
+  this.scope['activeAreas'] = this.scope['areas'][0].getLabel() == 'No results' ? [] : this.activeAreas();
+
+  menuRoot.find(os.ui.AreasCtrl.EventType.ALL).enabled = (this.scope['areas'][0].getLabel() != 'No results');
+  menuRoot.find(os.ui.AreasCtrl.EventType.SELECTED).enabled = (this.scope['selected'].length > 0);
+  menuRoot.find(os.ui.AreasCtrl.EventType.ACTIVE).enabled = (this.scope['activeAreas'].length > 0);
+
+  if (menu && target && target.length) {
+    menu.open({all: this.scope['areas'], active: this.scope['activeAreas'], selected: this.scope['selected']}, {
```

Comment:
The lack of quotes here on the object keys is an example where the compiler might mess with you. This goes away with my above fix suggestion.